rust_src/src/coding.rs

```diff
@@ -98,4 +98,16 @@ pub fn encode_file_name(fname: LispStringRef) -> LispStringRef {
     unsafe { c_encode_file_name(fname.into()) }.into()
 }
 
+/// Implements DECODE_SYSTEM macro
+/// Decode the string `input_string` using the specified coding system
+/// for system functions, if any.
+pub fn decode_system(input_string: LispStringRef) -> LispStringRef {
+    let local_coding_system: LispObject = unsafe { globals.Vlocale_coding_system }.into();
+    if !local_coding_system.is_nil() {
```

Comment:
How about flipping this so it reads `if coding system is nil { input_string } else { code_convert_string_norecord(...) }`.
Does that read as well or better?

Reply:
Sounds legit to me, fixed now